core/common/src/main/java/alluxio/util/ConfigurationUtils.java

```diff
@@ -41,48 +42,50 @@
   private ConfigurationUtils() {} // prevent instantiation
 
   /**
-   * Loads properties from resource. This method will search Classpath for the properties file with
-   * the given resourceName.
+   * Loads properties from a resource.
    *
-   * @param resourceName filename of the properties file
+   * @param resource url of the properties file
    * @return a set of properties on success, or null if failed
    */
-  public static Properties loadPropertiesFromResource(String resourceName) {
-    Properties properties = new Properties();
-
-    InputStream inputStream =
-        Configuration.class.getClassLoader().getResourceAsStream(resourceName);
-    if (inputStream == null) {
-      return null;
-    }
-
+  @Nullable
+  public static Properties loadPropertiesFromResource(URL resource) {
     try {
-      properties.load(inputStream);
+      return loadProperties(resource.openStream());
```

Comment:
@aaudiber  post-merge comment: don't we need to close this stream? Perhaps a try-with-resource block will help here?